app/src/main/java/a3/com/convo/activities/LoginActivity.java

```diff
@@ -140,23 +151,24 @@ public void onCompleted(
                         try {
                             final ParseUser user = ParseUser.getCurrentUser();
                             // initialize empty likes array
-                            user.put("pageLikes", new ArrayList<String>());
+                            user.put(Constants.PARSE_PAGE_LIKES_KEY, new ArrayList<String>());
                             // convert Json object into Json array
-                            JSONArray likes = json_object.getJSONObject("likes").optJSONArray("data");
+                            JSONArray likes = json_object.getJSONObject(Constants.LIKES_KEY).optJSONArray(Constants.DATA_KEY);
 
+                            // for each page, add it to our server if it's not there and add it to the user's likes array
                             for (int i = 0; i < likes.length(); i++) {
                                 final JSONObject page = likes.optJSONObject(i);
-                                String id = page.optString("id");
+                                String id = page.optString(Constants.ID_KEY);
 
                                 if (existingPages.containsKey(id)) {
                                     // page already exists in Parse, so we just get the object id and add it to their likes array
-                                    user.add("pageLikes", existingPages.get(id));
+                                    user.add(Constants.PARSE_PAGE_LIKES_KEY, existingPages.get(id));
                                 } else {
-                                    // doesn't exist yet, so we add it to the server
-                                    String category = page.optString("category");
-                                    String name = page.optString("name");
-                                    String coverUrl = page.getJSONObject("cover").optString("source");
-                                    String profUrl = page.getJSONObject("picture").getJSONObject("data").optString("url");
+                                    // page doesn't exist yet, so we add it to the server
+                                    String category = page.optString(Constants.CATEGORY_KEY);
+                                    String name = page.optString(Constants.NAME);
+                                    String coverUrl = page.getJSONObject(Constants.COVER).optString(Constants.SOURCE);
+                                    String profUrl = page.getJSONObject(Constants.PICTURE).getJSONObject(Constants.DATA_KEY).optString(Constants.URL);
```

Comment:
I see that this line is being edited by both you and @alveerak . I would let one of you update this block with my comments mentioned in the other PR.